src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/filtering/TumorEvidenceFilter.java

```diff
@@ -4,27 +4,30 @@
 import org.broadinstitute.hellbender.engine.ReferenceContext;
 import org.broadinstitute.hellbender.tools.walkers.mutect.clustering.Datum;
 import org.broadinstitute.hellbender.tools.walkers.mutect.clustering.SomaticClusteringModel;
+import org.broadinstitute.hellbender.utils.IndexRange;
 import org.broadinstitute.hellbender.utils.MathUtils;
 import org.broadinstitute.hellbender.utils.variant.GATKVCFConstants;
 
-import java.util.Collections;
-import java.util.List;
-import java.util.Optional;
+import java.util.*;
 
-public class TumorEvidenceFilter extends Mutect2VariantFilter {
+public class TumorEvidenceFilter extends Mutect2AlleleFilter {
     @Override
     public ErrorType errorType() { return ErrorType.SEQUENCING; }
 
     @Override
-    public double calculateErrorProbability(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext) {
+    protected List<Double> calculateErrorProbabilityForAlleles(final VariantContext vc, final Mutect2FilteringEngine filteringEngine, ReferenceContext referenceContext)
+    {
         final double[] tumorLods = Mutect2FilteringEngine.getTumorLogOdds(vc);
         final int[] ADs = filteringEngine.sumADsOverSamples(vc, true, false);
-        final int maxIndex = MathUtils.maxElementIndex(tumorLods);
-        final int altCount = ADs[maxIndex + 1];
         final int totalCount = (int) MathUtils.sum(ADs);
+        SomaticClusteringModel model = filteringEngine.getSomaticClusteringModel();
 
-        return filteringEngine.getSomaticClusteringModel()
-                .probabilityOfSequencingError(new Datum(tumorLods[maxIndex], 0, 0, altCount, totalCount, SomaticClusteringModel.indelLength(vc, maxIndex)));
+        List<Double> altResults = new ArrayList<>();
+        // 0 is the correct value. problem with threshold
+        new IndexRange(0, tumorLods.length).forEach(i ->
```

Comment:
Why not
```
return new IndexRange(0, tumorLods.length).mapToDouble(i -> _____);
```

Reply:
this is what i had to do. can you think of a way to simplify?
`        return Arrays.stream(new IndexRange(0, tumorLods.length).mapToDouble(i ->
                model.probabilityOfSequencingError(new Datum(tumorLods[i], 0, 0, ADs[i+1], totalCount, SomaticClusteringModel.indelLength(vc, i)))))
                .boxed().collect(Collectors.toList());
`

Reply:
Ah, I see, because `IndexRange.mapToDouble` gives you a `double[]`, not a list.  In that case how about
```
return IntStream.range(0, tumorLods.length)
   .mapToObj(i -> new Datum(tumorLods[i], 0, 0, ADs[i+1], totalCount, SomaticClusteringModel.indelLength(vc, i)))
   .map(model::probabilityOfSequencingError)
   .collect(Collectors.toList());
```